### PR TITLE
Added custom download directory

### PR DESCRIPTION
**Pull request by Blatzar**
needs some serious testing
havent tested previous downloads yet but should work

### CODE REVIEW

**Comment by LagradOst** [comment]
Tested and backwards compatible?

File: app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt [hunk 1]
```diff
@@ -44,8 +45,10 @@ class DownloadFragment : Fragment() {
     }
 
     private fun setList(list: List<VisualDownloadHeaderCached>) {
-        (download_list?.adapter as DownloadHeaderAdapter?)?.cardList = list
-        download_list?.adapter?.notifyDataSetChanged()
+        main {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt:48]: Why is main needed here?

File: app/build.gradle [hunk 2]
```diff
@@ -148,4 +148,7 @@ dependencies {
     // Networking
     implementation "com.squareup.okhttp3:okhttp:4.9.1"
     implementation "com.squareup.okhttp3:okhttp-dnsoverhttps:4.9.1"
+
+    // Util to skip the URI file fuckery 🙏
+    implementation "com.github.tachiyomiorg:unifile:17bec43"
```
Comment (LagradOst) [inline app/build.gradle:153]: Will you fix this?

File: app/src/main/res/xml/settings.xml [hunk 3]
```diff
@@ -104,6 +104,11 @@
                 android:summary="@string/dns_pref_summary"
                 android:icon="@drawable/ic_baseline_dns_24">
         </Preference>
+        <Preference
+                android:key="@string/download_path_key"
+                android:title="@string/download_path_pref"
+                android:icon="@drawable/netflix_download">
```
Comment (LagradOst) [inline app/src/main/res/xml/settings.xml:110]: Will this be wack on flashbang mode?

**Review by LagradOst — COMMENTED** [review]
LGTM

**Comment by Blatzar** [comment]
😔 Remember it can't play downloaded files all the time yet

**Pull request merged**